root_orchestrator/resource-abstractor/tests/api_test.py

```diff
@@ -0,0 +1,36 @@
+import json
+import unittest
+from unittest.mock import patch
+
+from api.v1.apps_blueprint import applicationsblp
+from bson import ObjectId
+from flask import Flask
+
+
+class BlueprintTestCase(unittest.TestCase):
+    def setUp(self):
+        self.app = Flask(__name__)
+        self.app.register_blueprint(applicationsblp)
+        self.client = self.app.test_client()
+
+    @patch("api.v1.apps_blueprint.apps_db.find_apps")
```

Comment:
Question regarding how this works (note: I am not new to writing tests but it has been a while):
Do I understand this correctly?:
- You want to check the "find_apps" method int he "apps_blueprints.apps.db" API.
- You mock this method - thus replacing it entirely - via the mock_find_apps which simply returns the mock return_value

If so how exactly does this test the API endpoint if it gets substituted by the mock entirely? Or does it only substitute the underlying DB interacting but the find_apps method is called thus properly (unit) tested?

Reply:
It is indeed a simple test. The mocked function is the db and not the endpoint entirely. So the endpoint is responsible in returning a json and converting the `_id` to a string from `ObjectId` so basically serialization is being tested here. Do you think it is still not really needed?

Reply:
If it is not mocking the entire thing and the endpoint code is executed this seems fine to me.